<launch>
    <arg name="skip_autoware_pose_initializer" default="false"/>

    <group if="$(var skip_autoware_pose_initializer)">
        <node name="gnss_pose_initializer_node" pkg="yabloc_pose_initializer" exec="gnss_pose_initializer_node" output="screen" args="--ros-args --log-level info">
            <param name="use_sim_time" value="$(var use_sim_time)"/>
            <remap from="ublox_topic" to="/sensing/gnss/ublox/navpvt"/>
            <remap from="pose_cov" to="/initialpose3d"/>
            <remap from="pf_pose" to="/localization/yabloc/pf/pose"/>
        </node> 
    </group>

    <group unless="$(var skip_autoware_pose_initializer)">
        <node name="camera_pose_initializer_node" pkg="yabloc_pose_initializer" exec="camera_pose_initializer_node" output="screen" args="--ros-args --log-level info">
            <param name="use_sim_time" value="$(var use_sim_time)"/>
            <param from="$(var camera_pose_initializer_param_path)" />
            <remap from="/image_raw" to="/localization/yabloc/imgproc/undistorted/image_raw"/>
            <remap from="input/camera_info" to="/localization/yabloc/imgproc/undistorted/camera_info"/>
            <remap from="initialpose" to="/initialpose"/>
            <remap from="/map/ll2_road_marking" to="/localization/yabloc/map/ll2_road_marking"/>
            <remap from="/ground_srv" to="/localization/yabloc/map/ground"/>
            <remap from="/semseg_srv" to="/localization/yabloc/semseg_srv"/>
            <remap from="yabloc_align_srv" to="/localization/yabloc/initializer/yabloc_align_srv"/>
        </node>

        <arg name="model_path" default="$(find-pkg-share yabloc_pose_initializer)/data/saved_model/model_float32.pb"/>
        <node name="semseg_node" pkg="yabloc_pose_initializer" exec="semseg_server.py" output="screen" args="--ros-args --log-level info">
            <param name="model_path" value="$(var model_path)"/>
            <remap from="/srv/semseg_srv" to="/localization/yabloc/semseg_srv"/>
        </node>
    </group>

</launch>
